src/scss/components/_app.scss

```diff
@@ -10,23 +10,83 @@
+  grid-column-gap: var(--of-app-grid-column-gap, 20px);
+  grid-template-columns: 2fr 1fr;
+  grid-template-areas:
+    'lang-switcher lang-switcher'
+    'body          progress-indicator'
+    'debug         debug';
   position: relative;
 
-  @include mobile-only {
-    padding-block-end: var(--of-form-mobile-padding-block-end, 15px);
-    padding-block-start: var(--of-form-mobile-padding-block-start, 0);
-    padding-inline-end: var(--of-form-mobile-padding-inline-end, 15px);
-    padding-inline-start: var(--of-form-mobile-padding-inline-start, 15px);
+  // When there's no progress indicator, stretch the main content
+  // over both containers.
+  @include bem.modifier('no-progress-indicator') {
+    @include bem.element('body') {
+      grid-row-start: body;
+      grid-row-end: progress-indicator;
+      grid-column-start: body;
+      grid-column-end: progress-indicator;
+    }
   }
 
   @include bem.element('language-switcher') {
+    grid-area: lang-switcher;
     display: flex;
   }
 
+  @include bem.element('body') {
+    grid-area: body;
+    padding-block-end: var(--of-app-body-padding-block-end, 0);
+    padding-block-start: var(--of-app-body-padding-block-start, 0);
+  }
+
+  @include bem.element('progress-indicator') {
+    grid-area: progress-indicator;
+  }
+
   @include bem.element('debug') {
+    grid-area: debug;
     margin-block-start: 2em;
   }
+
+  // Responsive styles - switch to a column layout and re-order elements.
+  @include mobile-only {
+    // https://stackoverflow.com/a/63609468 just 1fr doesn't work, but minmax does?
+    grid-template-columns: minmax(0, 1fr);
+    grid-template-areas:
+      'progress-indicator'
+      'lang-switcher'
+      'body'
+      'debug';
+    grid-row-gap: var(--of-app-mobile-grid-row-gap, 0);
+
+    padding-block-end: var(--of-app-mobile-padding-block-end, 15px);
+    padding-block-start: var(--of-app-mobile-padding-block-start, 0);
+    padding-inline-end: var(--of-app-mobile-padding-inline-end, 15px);
+    padding-inline-start: var(--of-app-mobile-padding-inline-start, 15px);
+
+    @include bem.element('body') {
+      padding-block-end: var(
+        --of-app-body-mobile-padding-block-end,
+        var(--of-app-body-padding-block-end, 0)
+      );
+      padding-block-start: var(
+        --of-app-body-mobile-padding-block-start,
+        var(--of-app-body-padding-block-start, 0)
+      );
+    }
+
+    @include bem.element('progress-indicator') {
+      margin-inline-end: var(--of-app-progress-indicator-mobile-margin-inline-end, -15px);
+      margin-inline-start: var(--of-app-progress-indicator-mobile-margin-inline-start, -15px);
+
+      // on mobile, the order of elements is swapped and to keep the progress indicator
+      // in view, we need to apply the positioning to this element rather than
+      // .openforms-progress-indicator
+      position: sticky;
+      inset-block-start: var(--of-app-progress-indicator-mobile-inset-block-start, 0);
+      z-index: 1;
```

Comment:
This is probably not right yet - when doing an appointment form the datepicker hides behind the progress indicator.